clarity_ext/domain/common.py

```diff
@@ -1,10 +1,40 @@
+from __builtin__ import isinstance
+
+
 class DomainObjectMixin(object):
 
     def __eq__(self, other):
         if isinstance(other, self.__class__):
-            return self.__dict__ == other.__dict__
+            return self._eq_rec(self, other)
         else:
             return False
 
+    def _eq_rec(self, a, b, cache=[]):
```

Comment:
Interesting. I guess that another way to solve this is to not use __dict__ comparison, but instead write an equality operator for each domain object. I think that might be cleaner anyways. But I think going for the generic solution is fine for now at least.